biorxiv/word_vector_experiment/biorxiv_visualize_embeddings.py

```diff
@@ -0,0 +1,362 @@
+
+# In[1]:
+
+
+import itertools
+from pathlib import Path
+import re
+
+import pandas as pd
+import plotnine as p9
+
+
+# In[2]:
+
+
+journal_map_df = pd.read_csv("../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t")
+journal_map_df.head()
+
+
+# In[3]:
+
+
+biorxiv_umap_models = {
+    int(re.search(r"(\d+)", str(file)).group(1)):pd.read_csv(str(file), sep="\t")
+    for file in Path().rglob("output/embedding_output/umap/biorxiv_umap*tsv")
+}
+
+# Keep only current versions of articles
+biorxiv_umap_models_lastest = {
+    key: (
+        biorxiv_umap_models[key]
+        .groupby("doi")
+        .agg({
+            "doi": "last",
+            "document": "last",
+            "umap1": "last",
+            "umap2": "last",
+            "category":"last",
+        })
+        .reset_index(drop=True)
+    )
+    for key in biorxiv_umap_models
+}
+
+
+# In[4]:
+
+
+biorxiv_tsne_models = {
+    int(re.search(r"(\d+)", str(file)).group(1)):pd.read_csv(str(file), sep="\t")
+    for file in Path().rglob("output/embedding_output/tsne/biorxiv_tsne*tsv")
+}
+
+# Keep only current versions of articles
+biorxiv_tsne_models_lastest = {
+    key: (
+        biorxiv_tsne_models[key]
+        .groupby("doi")
+        .agg({
+            "doi": "last",
+            "document": "last",
+            "tsne1": "last",
+            "tsne2": "last",
+            "category":"last",
+        })
+        .reset_index(drop=True)
+    )
+    for key in biorxiv_umap_models
+}
+
+
+# # UMAP of the Documents
+
+# In[5]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[150])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(
+        title="UMAP of BioRxiv (150 dim)",
+        color="Article Category"
+    )
+)
+g.save("output/embedding_output/umap/figures/biorxiv_umap_150.png", dpi=500)
+print(g)
+
+
+# In[6]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[250])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(
+        title="UMAP of BioRxiv (250 dim)",
+        color="Article Category"
+    )
+)
+g.save("output/embedding_output/umap/figures/biorxiv_umap_250.png", dpi=500)
+print(g)
+
+
+# In[7]:
+
+
+g = (
+    p9.ggplot(biorxiv_umap_models_lastest[300])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(
+        title="UMAP of BioRxiv (300 dim)",
+        color="Article Category"
+    )
+)
+g.save("output/embedding_output/umap/figures/biorxiv_umap_300.png", dpi=500)
+print(g)
+
+
+# Looking at the plots it seems that documents in the same category mostly cluster together, which is expected. The plots appear to be rotated versions of each other, which suggests that dimensionality doesn't hugely affect umap's results. Interesting cases are the outliers that appear within the plot. Question that arises: what are those outliers?
```

Comment:
Really cool! I wonder if there's anyway to do some sort of compression step that might help to distinguish groups more. So group combinations of words and then do this clustering?

Reply:
>I wonder if there's anyway to do some sort of compression step that might help to distinguish groups more. So group combinations of words and then do this clustering?

Hmmm.. I thought that's what tSNE and UMAP are suppose to do for datapoints (documents in my context). Am I missing something here?

Reply:
PCA might be more interpretable... no guarantees of course. Did you try it?

Reply:
Haven't tried it yet, but will do now.